src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,26 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserValidationException;
 import core.basesyntax.model.User;
+import core.basesyntax.validation.UserValidator;
+import core.basesyntax.validation.UserValidatorImpl;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
+    private final UserValidator userValidator = new UserValidatorImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        try {
+            userValidator.validate(user);
+            if (storageDao.get(user.getLogin()) != null) {
+                throw new UserValidationException("User with this login already exists");
+            }
+            storageDao.add(user);
+        } catch (UserValidationException exception) {
+            exception.printStackTrace();
+        }
+        return user;
```

Comment:
Returning the user object after catching an exception can lead to incorrect behavior, as mentioned previously. If the registration fails due to validation, the method should not return the user object as if it was successful.